src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts

```diff
@@ -35,23 +40,21 @@ export class OssVulnerabilityCountService implements Disposable {
     private readonly workspace: IVSCodeWorkspace,
     private readonly window: IVSCodeWindow,
     private readonly languages: IVSCodeLanguages,
-    private readonly vulnerabilityCountProvider: ModuleVulnerabilityCountProvider,
+    private readonly vulnerabilityCountProvider: OssVulnerabilityCountProvider,
     private readonly ossService: OssService,
     private readonly logger: ILog,
     private readonly editorDecorator: EditorDecorator,
     private readonly analytics: IAnalytics,
     private readonly configuration: IConfiguration,
   ) {}
-
   activate(): boolean {
     this.disposables.push(
-      (this.diagnostics = this.languages.createDiagnosticCollection(DIAGNOSTICS_OSS_COLLECTION_NAME)),
-      this.workspace.onDidChangeTextDocument(ev => {
-        if (ev.contentChanges.length) {
-          this.processFile(ev.document);
+      this.workspace.onDidChangeTextDocument((ev: TextDocumentChangeEvent) => {
+        if (ev?.contentChanges.length) {
+          _.debounce(() => this.processFile(ev.document), 500);
```

Comment:
suggestion: I think I'm seeing problems with the rendering of the decorator with this debounce. Can we consider removing it?

Reply:
![image](https://github.com/snyk/vscode-extension/assets/20150761/b82d73d1-a196-47b4-b069-13dafdc264dd)

Reply:
hmm, this should not happen :). working on a fix.